Realm/RLMQueryUtil.mm

```diff
@@ -457,8 +605,13 @@ void update_query_with_value_expression(RLMSchema *schema,
     }
 
     validate_property_value(prop, value, @"Expected object of type %@ for property '%@' on object of type '%@', but received: %@", desc, keyPath);
-    add_constraint_to_query(query, prop.type, pred.predicateOperatorType,
-                            pred.options, indexes, index, value);
+    if (pred.leftExpression.expressionType == NSKeyPathExpressionType) {
+        add_constraint_to_query(query, prop.type, pred.predicateOperatorType,
+                                pred.options, indexes, index, value);
+    } else {
```

Comment:
shouldn't this check that `pred.rightExpression.expressionType == NSKeyPathExpressionType`?